### PR TITLE
Remove Coinbase maturity limitation from Tapyrus.

### PR DESCRIPTION
**Pull request by Naviabheeman**
Removed Coinbase maturity definition and usage from transaction validation code.
As a result tapyrus wallet does not need to remember immature balance. 
Updated unit tests to create chain of length "5" instead of 100. (some tests needed multiple outputs)
Removed all coinbase maturing code from functional tests i.e create only one block (not 100) to get a spendable coinbase output.
Updated wallet balances in some functional tests - as blocks rewards mature immediately there are tests where wallet balances had to take into account the transaction fee and block reward.

### CODE REVIEW

File: src/federationparams.cpp [hunk 1]
```diff
@@ -159,6 +159,8 @@ CPubKey CFederationParams::ReadAggregatePubkey(const std::vector<unsigned char>&
 
 bool CFederationParams::ReadGenesisBlock(std::string genesisHex)
 {
+    ECCVerifyHandle globalVerifyHandle;
+
```
Comment (azuchi) [inline src/federationparams.cpp]: What is this use for?

File: src/interfaces/wallet.h [hunk 2]
```diff
@@ -321,7 +320,7 @@ struct WalletBalances
     bool balanceChanged(const WalletBalances& prev) const
     {
         return balance != prev.balance || unconfirmed_balance != prev.unconfirmed_balance ||
-               immature_balance != prev.immature_balance || watch_only_balance != prev.watch_only_balance ||
+               watch_only_balance != prev.watch_only_balance ||
                unconfirmed_watch_only_balance != prev.unconfirmed_watch_only_balance ||
                immature_watch_only_balance != prev.immature_watch_only_balance;
```
Comment (azuchi) [inline src/interfaces/wallet.h]: Is `immature_watch_only_balance` still necessary?

File: src/federationparams.cpp (hunk 1 above)
Comment (Naviabheeman) [reply src/federationparams.cpp]: It was needed in unit tests for initialising verification context of signature verification. IT should be in the constructor so I removed from here.

File: src/interfaces/wallet.h (hunk 2 above)
Comment (Naviabheeman) [reply src/interfaces/wallet.h]: no, I removed it

**Pull request merged**